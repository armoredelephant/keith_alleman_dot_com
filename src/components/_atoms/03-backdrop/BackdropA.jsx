import styled from 'styled-components';

const BackdropA = styled.div`
    position: fixed;
    top: 0;
    right: 0;
    height: 100%;
    background: rgba(0, 0, 0, 0.3);
    width: 0;
    opacity: 0;
    visibility: hidden;
    transition-property: all;
    transition-duration: 1s;
    transition-timing-function: cubic-bezier(0.19, 1, 0.22, 1);
    transition-delay: 0.5s;  

    .active & {
        width: 100%;
        visibility: visible;
        opacity: 1;
        transition-property: all;
        transition-duration: 2s;
        transition-timing-function: cubic-bezier(0.19, 1, 0.22, 1);
        transition-delay: 0.5s; 
    }
`;

export default BackdropA;