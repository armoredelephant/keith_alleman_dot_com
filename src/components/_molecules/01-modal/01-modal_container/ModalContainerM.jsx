import React from 'react';

import BackdropA from '../../../_atoms/03-backdrop/BackdropA.jsx';
import ModalM from '../00-modal/ModalM.jsx';

const ModalContainerM = ( props ) => {
    const handleBackgroundClick = e  => {
        if (e.target === e.currentTarget) props.handleModal();
    };

    return (
        <React.Fragment>
            <BackdropA {...props} onClick={e => handleBackgroundClick(e)} />
            <ModalM />
        </React.Fragment>
    );
}

export default ModalContainerM;