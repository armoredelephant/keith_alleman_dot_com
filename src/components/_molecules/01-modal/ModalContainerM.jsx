import React from 'react';

/**
 * (149,213,210) - 95D5D2 teal
 * (250,174,96) - FAAE60 - orange
 * (153,229,178) - 99E5B2 - mint
 * (89,79,79) - 594F4F - gray
 * (152,164,196) - 98A4C4 - purple/blue
 */

import BackdropA from '@A/03-backdrop/BackdropA';
import ModalM from './ModalM';

const ModalContainerM = () => {
  return (
    <React.Fragment>
      <BackdropA />
      <ModalM />
      {/** container for nav within modal */}
      {/** two sections, upper section and bottom section? */}
      {/* background will be it's own? */}
    </React.Fragment>
  );
};

export default ModalContainerM;
