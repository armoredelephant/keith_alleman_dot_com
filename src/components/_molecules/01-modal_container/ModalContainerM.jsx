import React from 'react';

import BackdropA from '../../_atoms/03-backdrop/BackdropA';
import ModalA from '../../_atoms/04-modal/ModalA';

const ModalContainerM = ( props ) => {
    const handleBackgroundClick = e  => {
        if (e.target === e.currentTarget) props.handleModal();
    };

    return (
        <React.Fragment>
            <BackdropA onClick={e => handleBackgroundClick(e)} />
            <ModalA />
        </React.Fragment>
    );
}

export default ModalContainerM;