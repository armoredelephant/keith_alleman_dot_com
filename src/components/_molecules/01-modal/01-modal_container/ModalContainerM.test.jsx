import React, { useState } from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';

import ModalContainerM from './ModalContainerM.jsx';
import BackdropA from '../../../_atoms/03-backdrop/BackdropA.jsx';

describe('ModalContainerM', () => {
    it('renders', () => {
        const wrapper = shallow(<ModalContainerM />);
        expect(wrapper).toBeTruthy();
    });

    it('renders correctly snapshot', () => {
        const output = shallow(<ModalContainerM />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('should handle the click event', () => {
        let activeModal = false;
        window.alert = jest.fn();
        const wrapper = shallow(<ModalContainerM {...props} activeModal='false' 
            handleModal={() => { return activeModal = !activeModal }} />);
        wrapper.find(BackdropA).simulate('click', { target: { BackdropA } });
        expect(window.alert).toHaveBeenCalledWith('clicked');
    });
});
