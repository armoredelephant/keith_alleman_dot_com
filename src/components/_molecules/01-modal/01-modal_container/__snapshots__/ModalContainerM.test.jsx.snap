// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContainerM renders correctly snapshot 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
  />
  <styled.div />
</Fragment>
`;
